NEWS.md

```diff
@@ -1,5 +1,7 @@
 # evaluate (development version)
 
+* The `source` output handler is now parsed the entire top-level expression, not just the first component.
+* `evaluate()` will now terminate on the first error in a top-level expression. This matches R's own behaviour more closely.
```

Comment:
I think I might have actually changed this behaviour when I started using restarts, but it's now tested and documented. This is a bug that umpire works around in https://github.com/rstudio/umpire/blob/main/R/evaluate.R#L6-L11.

Reply:
I'd say `1;2` is one top-level expression consisting of two expressions. In your definition, what's the difference between a TLE and an expression?

I'd say each TLE generates one source statement.

Reply:
Whether an expression is top-level is a property of where it's evaluated, at the top-level evaluation loop. It certainly makes sense to call `1;2` "top-level" but I find it confusing to also call it an "expression" because it's not an R expression stricto sensu. An expression is something that can be evaluated and thus must be representable as an AST node or leaf.

You could argue that `1;2` is parsed as an EXPRSXP vector and that you can evaluate it with the R-level `eval()` function, but I think it's the C-level function that should guide meaning here. And for the C-level function, EXPRSXP is a literal.

From this point of view `foo(bar)` consists of two expressions with `bar` nested in `foo(bar)`. Whereas `1; 2` is not an expression but a sequence of two expressions managed by a top-level evaluation loop.

> I'd say each TLE generates one source statement.

Sorry I'm not sure what that means.

Reply:
Ok, I see where you're coming from. I'm going to merge this PR but I'll keep thinking about the vocab.